theano/tensor/basic.py

```diff
@@ -4372,7 +4372,7 @@ def infer_shape(self, node, ishapes):
                     if ele == -1:
                         requ[i] = missing
             elif crit == 1:  # we reshape to -1
-                requ = [mul(*ishapes[0])]
+                requ = [mul(*ishapes[0])] if ishapes[0] else [1]
```

Comment:
By changing the interface you mean ishapes's interface or Reshape.infer_shape return's interface? Reshape.infer_shape is broken anyway and is Sina's next CCW ticket.

How does infer_shape usually process scalar dimensions in ishapes and in return?